objects/src/accounts/storage/mod.rs

```diff
@@ -267,193 +82,80 @@ impl AccountStorage {
-    /// - The update has a value arity different from 0.
-    pub fn set_map_item(
-        &mut self,
-        index: u8,
-        key: Word,
-        value: Word,
-    ) -> Result<(Word, Word), AccountError> {
-        // layout commitment slot cannot be updated
-        if index == Self::SLOT_LAYOUT_COMMITMENT_INDEX {
-            return Err(AccountError::StorageSlotIsReserved(index));
-        }
-
-        // only map slots of basic arity can currently be updated
-        match self.layout[index as usize] {
-            StorageSlotType::Map { value_arity } => {
-                if value_arity > 0 {
-                    return Err(AccountError::StorageSlotInvalidValueArity {
-                        slot: index,
-                        expected: 0,
-                        actual: value_arity,
-                    });
-                }
-            },
-            slot_type => Err(AccountError::MapsUpdateToNonMapsSlot(index, slot_type))?,
-        }
-
-        // get the correct map
-        let storage_map =
-            self.maps.get_mut(&index).ok_or(AccountError::StorageMapNotFound(index))?;
-
-        // get old map root to return
-        let old_map_root = storage_map.root();
-
-        // update the key-value pair in the map
-        let old_value = storage_map.insert(key.into(), value);
-
-        // update the root of the storage map in the corresponding storage slot
-        let index = LeafIndex::new(index.into()).expect("index is u8 - index within range");
-        self.slots.insert(index, storage_map.root().into());
-
-        Ok((old_map_root.into(), old_value))
+        self.slots[index as usize] = storage_slot;
     }
 }
 
-// UTILITIES
+// HELPER FUNCTIONS
 // ------------------------------------------------------------------------------------------------
 
-/// Computes the commitment to the given layout
-fn layout_commitment(layout: &[StorageSlotType]) -> Digest {
-    Hasher::hash_elements(&layout.iter().map(Felt::from).collect::<Vec<_>>())
+/// Converts given slots into field elements
+fn slots_as_elements(slots: &[StorageSlot]) -> Vec<Felt> {
+    slots
+        .iter()
+        .flat_map(|storage_slot| {
+            let mut elements: Vec<Felt> = Vec::with_capacity(8);
+            match storage_slot {
+                StorageSlot::Value(word) => {
+                    for element in word {
+                        elements.push(*element)
+                    }
+                    for _ in 0..4 {
+                        elements.push(ZERO)
+                    }
+                },
+                StorageSlot::Map(map) => {
+                    let smt_root = map.root();
+
+                    for element in smt_root.as_elements() {
+                        elements.push(*element)
+                    }
+
+                    elements.push(ONE);
+
+                    for _ in 0..3 {
+                        elements.push(ZERO)
+                    }
+                },
+            }
```

Comment:
If we have methods mentioned in the previous comment, this could be simplified quite a bit. For example:

```rust
elements.extend_from_slice(&slot.to_word());
elements.extend_from_slice(&slot.get_type().as_word());
```